fastapi/applications.py

```diff
@@ -134,6 +136,36 @@ def __init__(
         self.openapi_schema: Optional[Dict[str, Any]] = None
         self.setup()
 
+    def build_middleware_stack(self) -> ASGIApp:
+        # Duplicate/override from Starlette to add AsyncExitStackMiddleware after
+        # inside of ExceptionMiddleware
+        debug = self.debug
+        error_handler = None
+        exception_handlers = {}
+
+        for key, value in self.exception_handlers.items():
+            if key in (500, Exception):
+                error_handler = value
+            else:
+                exception_handlers[key] = value
+
+        middleware = (
+            [Middleware(ServerErrorMiddleware, handler=error_handler, debug=debug)]
+            + self.user_middleware
+            + [
+                Middleware(
+                    ExceptionMiddleware, handlers=exception_handlers, debug=debug
+                ),
+                # Add FastAPI-specific AsyncExitStackMiddleware for dependencies
+                Middleware(AsyncExitStackMiddleware),
+            ]
+        )
+
+        app = self.router
+        for cls, options in reversed(middleware):
```

Comment:
Rename `cls` to `klass`, `cls` is usually used to refer to the current class. This is a convention. [I encourage you to go ahead.](https://github.com/django/django/blob/7119f40c9881666b6f9b5cf7df09ee1d21cc8344/django/contrib/gis/sitemaps/views.py#L18)

Reply:
Thanks for your interest. But I prefer to keep the code as-is, the same way it is in Starlette.